tests/bluetooth/bsim_bt/bsim_test_mesh/src/test_advertiser.c

```diff
@@ -51,6 +55,58 @@ static void adv_init(void)
 	ASSERT_OK(bt_mesh_adv_enable(), "Mesh adv init failed");
 }
 
+static void allocate_all_net_buf(struct net_buf **buf, size_t num_buf, uint8_t xmit)
+{
+	for (int i = 0; i < num_buf; i++) {
+		*buf = bt_mesh_adv_create(BT_MESH_ADV_DATA, BT_MESH_LOCAL_ADV,
+					  xmit, K_NO_WAIT);
+
+		ASSERT_FALSE(!*buf, "Out of buffers");
+		buf++;
+	}
+}
+
+static void verify_net_buf_overflow(void)
+{
+	struct net_buf *dummy_buf;
+
+	/* Verity Queue overflow */
+	dummy_buf = bt_mesh_adv_create(BT_MESH_ADV_DATA, BT_MESH_LOCAL_ADV,
+				       BT_MESH_TRANSMIT(2, 20), K_NO_WAIT);
+	ASSERT_TRUE(!dummy_buf, "Unexpected extra buffer");
+}
+
+static bool check_retr_delta_time(void)
+{
+	static int cnt;
+	static int64_t timestamp;
+	static int64_t delta;
+
+	if (cnt) {
+		delta = k_uptime_delta(&timestamp);
+
+		LOG_INF("rx: cnt(%d) delta(%dms)", cnt, delta);
+
+		ASSERT_TRUE(delta >= xmit_param.interval &&
+			    delta < (2 * xmit_param.interval));
```

Comment:
Same as here: https://github.com/zephyrproject-rtos/zephyr/pull/50195#discussion_r977857124
I guess there is no need in a separate callback and the same code could be used by other tests.

Reply:
I guess to reach the same precision you have to migrate the common test configuration on system clock 32768 as I did in my PR (the stability of other tests should be considered) or move these tests to the GATT configuration.

Reply:
@Balaklaka, please pay attention to my comment above. It will help you to reach a precision of better than 15 ms.